lightning/src/chain/channelmonitor.rs

```diff
@@ -2451,7 +2451,8 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 					output: outp.clone(),
 				});
 				break;
-			} else if let Some(ref broadcasted_holder_revokable_script) = self.broadcasted_holder_revokable_script {
+			}
+			if let Some(ref broadcasted_holder_revokable_script) = self.broadcasted_holder_revokable_script {
 				if broadcasted_holder_revokable_script.0 == outp.script_pubkey {
 					spendable_output =  Some(SpendableOutputDescriptor::DelayedPaymentOutput(DelayedPaymentOutputDescriptor {
```

Comment:
Added some print statements and it seems like the issue was: it'd enter this `else if` statement, but wouldn't pass the check on the second highlighted line, and then naturally fail to even consider the next `else if` check

Reply:
Given the second if clause overwrites spendable_output, it seems like essentially a reordering of the clauses?

Reply:
That might have worked too, but it's really the if else blocks miss that one in the middle isn't fully captured by the check. Because the check is split across the if let and the inner if, we sometimes incorrectly short-circuit the if else tree.

Reply:
Yeah, I was wondering if they could have possibly been merge-able into one condition.

Reply:
No, rather obnoxiously Rust doesn't let you merge if let into any additional constraints.

Reply:
Here my understanding of the fixed issue.

If a holder commitment is broadcast, we set `ChannelMonitor::broadcasted_holder_revokable_script` to `Some` in `check_spend_holder_transaction`, from then always marking the spendable output as a `DelayedPaymentOutput` in `is_paying_spendable_output`.

After this PR, the spendable output type detection is moved from a "else if" control flow to a "if" one.